chore!: Use Jackson for all JSON deserialization in codecs

Artur- opened the pull request:
Simplify JacksonCodec and JsonCodec by using Jackson for all type
deserialization instead of custom primitive type handling. This provides
consistent behavior and better error messages but removes lenient type
conversions (e.g., boolean to integer).

caalador commented:
```
PublishedServerEventHandlerRpcHandlerTest.nullValueAreAcceptedForPrimitive:
Cannot deserialize JSON to type int: Cannot map `null` into type `int` (set DeserializationConfig.DeserializationFeature.FAIL_ON_NULL_FOR_PRIMITIVES to 'false' to allow)
```

`JavaScriptReturnValueIT.testAllCombinations` For this I guess null is the breaking one that the doesn't execute due to an error.

Artur- commented:
Fixed PublishedServerEventHandlerRpcHandlerTest

The `JavaScriptReturnValueIT.testAllCombinations` tried returning `new Error('message')` from JS, which is serialized as `{}` and then using that as a string. It makes no sense, so I removed it
